lock-api/src/main/java/com/palantir/lock/client/LockTokenShare.java

```diff
@@ -60,7 +65,7 @@ private LockTokenShare(ReferenceCounter referenceCounter, LockToken token) {
      * @return referenced shared lock token if all lock token shares are unlocked; that is lock token on server side is
      * good to be unlocked.
      */
-    public synchronized Optional<LockToken> unlock() {
+    synchronized Optional<LockToken> unlock() {
```

Comment:
noted this in the PR that added this class, but I still find this odd. There's levels of synchronization that imply single-threaded-ness; that only one thread will ever be returned the lock token to unlock.

However, there's a race and this could _possibly_ happen twice if the race is hit. And, unlock is not idempotent, which is also weird; it means that the person abusing this class could see different behaviour.

As before, I'd prefer our reference counter return a single share the returned token to unlock; the last person to unlock is handed the token to unlock and the race goes away.

Does this make sense?